main.go

```diff
@@ -3,81 +3,98 @@ package main
 import (
 	"context"
 	"flag"
+	"fmt"
 	"log"
-	"net"
 
 	"github.com/armon/go-socks5"
 )
 
 const (
-	allowed = true
-	denied  = false
+	allowed            = true
+	denied             = false
+	letsEncryptProd    = "acme-v02.api.letsencrypt.org"
+	letsEncryptStaging = "acme-staging-v02.api.letsencrypt.org"
 )
 
-type myRule struct {
-	addrs []net.IP
-	fqdns []string
+type allowLetsEncrypt struct{}
+
+func (m allowLetsEncrypt) Allow(ctx context.Context, req *socks5.Request) (context.Context, bool) {
+	for _, fqdn := range []string{letsEncryptProd, letsEncryptStaging} {
+		if req.DestAddr.FQDN == fqdn {
+			return ctx, allowed
+		}
+	}
+	return ctx, denied
+}
+
+type logAll struct {
+	rules socks5.RuleSet
 }
 
-func logConn(allowed bool, from, to *socks5.AddrSpec) {
-	var prefix string
+func (m logAll) Allow(ctx context.Context, req *socks5.Request) (context.Context, bool) {
+	ctx, allowed := m.rules.Allow(ctx, req)
+
+	prefix := "Denying"
 	if allowed {
 		prefix = "Allowing"
-	} else {
-		prefix = "Denying"
 	}
+	from := req.RemoteAddr
+	to := req.DestAddr
 	log.Printf("%s connection request from %s:%d (%s) to %s:%d (%s).",
 		prefix,
 		from.IP, from.Port, from.FQDN,
-		to.IP, to.Port, to.FQDN)
+		to.IP, to.Port, to.FQDN,
+	)
+
+	return ctx, allowed
 }
 
-func (m myRule) Allow(ctx context.Context, req *socks5.Request) (context.Context, bool) {
-	for _, addr := range m.addrs {
-		if req.DestAddr.IP.Equal(addr) {
-			logConn(allowed, req.RemoteAddr, req.DestAddr)
-			return ctx, allowed
-		}
-	}
-	for _, fqdn := range m.fqdns {
-		if req.DestAddr.FQDN == fqdn {
-			logConn(allowed, req.RemoteAddr, req.DestAddr)
-			return ctx, allowed
-		}
-	}
-	logConn(denied, req.RemoteAddr, req.DestAddr)
-	return ctx, denied
+type config struct {
```

Comment:
after adding a few more cli-args, it seemed easier to add in a struct to manage the info from the cli.